# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/HostWebGLContext.h
# Commit: 0b191b707750
# Full Hash: 0b191b707750e040289033359306e4966c1bf062
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r 16c5dd2cfd60 -r 0b191b707750 dom/canvas/HostWebGLContext.h
--- a/dom/canvas/HostWebGLContext.h	Mon Jun 15 18:25:55 2020 +0000
+++ b/dom/canvas/HostWebGLContext.h	Mon Jun 15 18:25:58 2020 +0000
@@ -184,9 +184,11 @@
     mContext->SetCompositableHost(compositableHost);
   }
 
-  void Present() { return (void)mContext->Present(); }
-  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(
-      const layers::TextureType t) const;
+  void Present(const ObjectId fb, const layers::TextureType t) const {
+    return (void)mContext->Present(AutoResolve(fb), t);
+  }
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(ObjectId fb,
+                                                  layers::TextureType) const;
 
   RefPtr<gfx::DataSourceSurface> GetFrontBufferSnapshot() const {
     return mContext->GetFrontBufferSnapshot();